pubsub.go

```diff
@@ -42,12 +39,43 @@ type PubSub struct {
 
 	cmd *Cmd
 
+	size int
+
 	chOnce sync.Once
 	msgCh  chan *Message
 	allCh  chan interface{}
 	ping   chan struct{}
+
+	pingTimeout   time.Duration
+	healthTimeout time.Duration
+}
+
+type PubSubOption func(c *PubSub)
```

Comment:
PubSubOption -> ChannelOption since this changes `Channel` behaviour, not PubSub